projects/hipsparse/CMakeLists.txt

```
@@ -1,289 +1,278 @@
+    # if(HIPSPARSE_ENABLE_FORTRAN)
+        # message(FATAL_ERROR "Cannot enable fortran clients and code coverage.")
+    if((NOT CMAKE_BUILD_TYPE STREQUAL "Debug") AND (NOT CMAKE_BUILD_TYPE STREQUAL "RelWithDebInfo"))
+        message(FATAL_ERROR "Code coverage is only supported for CMAKE_BUILD_TYPE=Debug and CMAKE_BUILD_TYPE=RelWithDebInfo.") # rel with deb info also
+    endif()
 endif()
 
-# Build options
-option(BUILD_SHARED_LIBS "Build hipSPARSE as a shared library" ON)
-option(BUILD_CLIENTS_TESTS "Build tests (requires googletest)" OFF)
-option(BUILD_CLIENTS_BENCHMARKS "Build benchmarks" ON)
-option(BUILD_CLIENTS_SAMPLES "Build examples" ON)
-option(BUILD_CLIENTS_ONLY "Build only clients" OFF)
-option(BUILD_VERBOSE "Output additional build information" OFF)
-option(USE_CUDA "Build hipSPARSE using CUDA backend" OFF)
-option(BUILD_CUDA "Build hipSPARSE using CUDA backend" OFF)
-option(BUILD_CODE_COVERAGE "Build with code coverage enabled" OFF)
-option(BUILD_ADDRESS_SANITIZER "Build with address sanitizer enabled" OFF)
-option(BUILD_DOCS "Build documentation" OFF)
-
-if(BUILD_CUDA)
-  message(DEPRECATION "Using BUILD_CUDA is deprecated and will be removed as an option in a future release. Instead use USE_CUDA")
-  set(USE_CUDA ${BUILD_CUDA})
+if(HIPSPARSE_ENABLE_ASAN AND HIPSPARSE_ENABLE_FORTRAN)
+    message(FATAL_ERROR "Cannot enable both address sanitizer and Fortran clients")
 endif()
 
-if(BUILD_CODE_COVERAGE)
-  add_compile_options(-fprofile-arcs -ftest-coverage)
-  add_link_options(--coverage)
+# Set default build type
+if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
+    message(STATUS "Setting build type to 'Release' as none was specified.")
+    set(CMAKE_BUILD_TYPE "Release" CACHE STRING "Choose the type of build." FORCE)
+    set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "" "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
 endif()
 
-if(BUILD_ADDRESS_SANITIZER)
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsanitize=address -shared-libasan")
-  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fsanitize=address -shared-libasan")
-  add_link_options(-fuse-ld=lld)
+# Conditionally enable Fortran
+# NOTE: Behavior change - Fortran now controlled by HIPSPARSE_ENABLE_FORTRAN option (default ON)
+# Previously enabled automatically when building tests or benchmarks
+if(HIPSPARSE_ENABLE_FORTRAN)
+    enable_language(Fortran)
 endif()
 
-# Dependencies
-include(cmake/Dependencies.cmake)
+# --- Dependency Setup ---
 
-# Setup version
-rocm_setup_version(VERSION 4.2.0)
-set(hipsparse_SOVERSION 4.2.0)
+find_package(Python3 3.9 COMPONENTS Interpreter REQUIRED)
 
-# hipSPARSE library
-if (NOT BUILD_CLIENTS_ONLY)
-  add_subdirectory(library)
+if(HIPSPARSE_ENABLE_CUDA)
+    find_package(HIP REQUIRED CONFIG)
+    # list(APPEND HIP_INCLUDE_DIRS "${HIP_ROOT_DIR}/include")
+    find_package(CUDA REQUIRED)
+elseif(HIPSPARSE_ENABLE_HIP)
+    find_package(hip REQUIRED CONFIG)
+    find_package(rocsparse 4.0.1 REQUIRED CONFIG)
```

```suggestion
    find_package(rocsparse REQUIRED)
```

Is there a reason that we are providing a version?